### DISCUSSION TITLE
Automower heatmap

### DISCUSSION THREAD

**Discussion by celsohernando** [Thomas55555/husqvarna_automower#451] [Ideas]
HI, it would be nice to have a heatmap of the positions of the automwer for the last week (or a configurable period of time).
If you don't plan to implement this, how could  I programatically access the GPS positions over a configurable period of time ?
thanks

**Comment by robinostlund** [comment]
@celsohernando wouldn't it be possible to the map card together with the device_tracker? and then set hours to show to how many hours you would like to have?

**Comment by prairiesnpr** [comment]
I agree that this would be a nice feature, there isn't currently a great way to do this, since HA doesn't support polygonal zones.  I have a custom fork that allows me to track the time the mower spends in each zone, but it's really a work around for a limitation in HA. It's unlikely we will see it added to HA per discussion here. https://community.home-assistant.io/t/wth-we-only-have-round-shaped-zones/468137

![image](https://github.com/Thomas55555/husqvarna_automower/assets/18637008/86f92b5e-98f9-41ec-847c-40cc0a571180)

**Comment by celsohernando** [comment]
Hi, thanks @robinostlund @prairiesnpr for your comments. I was thinking in drawing  a 2D histogram with the GPS probes of the last month/week/day, filtered by status "mowing" and superimpose it to the image map of the integration, that has already the same reference system.
